trickster/endpoints/mocked.py

```diff
@@ -20,9 +20,13 @@ def mocked_response(request: Request, mocked_router: Router = Depends(get_router
     response = None
 
     if match := mocked_router.match(request):  # noqa: SIM102 - nested if statements
-        if matched_response := match.route.get_response(match):
-            match.route.hits += 1
-            response = matched_response
+        try:
+            match.route.authenticate(request)
+            if matched_response := match.route.get_response(match):
+                match.route.hits += 1
+                response = matched_response
+        except HTTPException:
```

Comment:
As Jakub already told you, you are converting any HTTPException to 401. Throw some your explicit exception `AuthenticationError` and translate it to 401.

Reply:
This issue is fixed here
https://github.com/TricksterOrg/trickster/pull/29/files#